### PR TITLE
Update keras layer BatchNormalization with synchronized argument

### PR DESCRIPTION
**Pull request by sineeli**
# Description

Existing `tf.keras.layers.experimental.SyncBatchNormalization` has been deprecated in **tensorflow-2.12.0**, so updated the `tf.keras.layers.BatchNormalization` with `synchronized` argument.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by sdhungan** [comment]
Why was this change not merged? And will the following error below not cause difference in functionality? 
"W0413 13:52:55.967910 140077427222336 batch_normalization.py:1426] `tf.keras.layers.experimental.SyncBatchNormalization` endpoint is deprecated and will be removed in a future release. Please use `tf.keras.layers.BatchNormalization` with parameter `synchronized` set to True."

**Comment by sineeli** [comment]
@sdhungan,

I will be fixing it again there was some problem while running the test cases.
